internal/test/e2e/migrations_test.go

```diff
@@ -109,8 +109,70 @@ func TestMigrations(t *testing.T) {
 			shardHosts += len(shard.Contracts)
 		}
 	}
+
 	// all shards should have 1 host except for 1. So we end up with 4 in total.
 	if shardHosts != 4 {
 		t.Fatalf("expected 4 shard hosts, got %v", shardHosts)
 	}
+
+	// create another bucket and add an object
+	tt.OK(b.CreateBucket(context.Background(), "newbucket", api.CreateBucketOptions{}))
+	tt.OKAll(w.UploadObject(context.Background(), bytes.NewReader(data), "newbucket", t.Name(), api.UploadObjectOptions{}))
+
+	// assert we currently don't have any alerts
+	ress, _ := b.Alerts(context.Background(), alerts.AlertsOpts{})
+	if ress.Totals.Error+ress.Totals.Critical != 0 {
+		t.Fatal("unexpected", ress)
+	}
+
+	// prepare
+
+	// remove all hosts to ensure migrations fail
+	for _, h := range cluster.hosts {
+		cluster.RemoveHost(h)
+	}
+
+	// fetch alerts and collect object ids until we found two
+	seen := make(map[types.Hash256]struct{})
```

Comment:
I think you can clean the `Retry` up a bit by dropping this and just setting `got = make(map[string][]string` at the beginning of the retry.